EvidenceApi/V1/UseCase/UpdateDocumentSubmissionVisibilityUseCase.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using System.Globalization;
+using System.Threading.Tasks;
+using EvidenceApi.V1.UseCase.Interfaces;
+using EvidenceApi.V1.Gateways.Interfaces;
+using EvidenceApi.V1.Boundary.Response;
+using EvidenceApi.V1.Boundary.Request;
+using EvidenceApi.V1.Boundary.Response.Exceptions;
+using EvidenceApi.V1.Domain;
+using EvidenceApi.V1.Factories;
+using EvidenceApi.V1.Domain.Enums;
+using Notify.Exceptions;
+using Microsoft.Extensions.Logging;
+
+namespace EvidenceApi.V1.UseCase
+{
+    public class UpdateDocumentSubmissionVisibilityUseCase : IUpdateDocumentSubmissionVisibiltyUseCase
+    {
+        private readonly IEvidenceGateway _evidenceGateway;
+        private readonly IDocumentTypeGateway _documentTypeGateway;
+        private readonly ILogger<UpdateDocumentSubmissionVisibilityUseCase> _logger;
+
+        public UpdateDocumentSubmissionVisibilityUseCase(
+            IDocumentTypeGateway documentTypeGateway,
+            IEvidenceGateway evidenceGateway,
+            ILogger<UpdateDocumentSubmissionVisibilityUseCase> logger
+        )
+        {
+            _documentTypeGateway = documentTypeGateway;
+            _evidenceGateway = evidenceGateway;
+            _logger = logger;
+        }
+
+        public DocumentSubmissionResponse ExecuteAsync(Guid id, DocumentSubmissionVisibilityUpdateRequest request)
+        {
+            var documentSubmission = _evidenceGateway.FindDocumentSubmission(id);
```

Comment:
If the document is already hidden this wil return null, so at the moment it's not possible to unhide a document. Probably worth reviewing this and adding support for both hiding and unhiding or modify the setup, so it clearly works only one way. I guess both way updates would make more sense. Just in case it's needed.

Reply:
I have created a new gateway for this purpose only, that will return all submission, regardless of the state.